sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -1712,6 +1712,20 @@ class CastSuite extends CastSuiteBase {
       assert(e3.contains("Casting 2147483648 to int causes overflow"))
     }
   }
+
+  test("SPARK-34902: Cast support DayTimeIntervalType and YearMonthIntervalType") {
```

Comment:
@AngersZhuuuu Could you add tests for corner cases like `null`, `Int/Long.MaxValue/MinValue`, `0`, please.

What happens in casting a `Int` to `DayTimeIntervalType`? Will it be automatically converted to `Long`?

Reply:
> @AngersZhuuuu Could you add tests for corner cases like `null`, `Int/Long.MaxValue/MinValue`, `0`, please.
> 
> What happens in casting a `Int` to `DayTimeIntervalType`? Will it be automatically converted to `Long`?

Int.MinValue is not return 0. So should I add logic when less then 0 throw exception or return 0?

Reply:
I don't fully understand the problem. Could you give me an example, please, and explain the issue.

Reply:
> I don't fully understand the problem. Could you give me an example, please, and explain the issue.

Current result 
```
checkEvaluation(
      cast(cast(0, IntegerType), DayTimeIntervalType), 0L)
    checkEvaluation(
      cast(cast(null, IntegerType), DayTimeIntervalType), null)
    checkEvaluation(
      cast(cast(Int.MinValue, IntegerType), DayTimeIntervalType), -2147483648L)
```

`DayTimeIntervalType` and `YearMonthIntervalType` should not be negetive, right?

Reply:
Why not, the intervals can be positive as well as negative. I think this is correct. I would just write it as:
```scala
checkEvaluation(cast(cast(Int.MinValue, IntegerType), DayTimeIntervalType), Int.MinValue.toLong)
```